### ISSUE TITLE
Better handling of builtins

### ISSUE THREAD

**Issue by xldenis** [xldenis/creusot#140]
Currently all builtin functions are handled through a giant switch which checks a bunch of defids. This is already unweidly but as we add more built in types (`Seq`) or more generally want to allow users to use other why3 libraries this is going to get worse and untenable. 

I think it might be possible to use Rust's extern mechanism for this thus giving our definitions a look like:

```
#[link_name("seq.Seq")]
extern "why3" {
  fn index(seq: Seq<T>, ix: Int) -> T;
  ...
}
```

**Comment by xldenis** [comment]
Ok, directly using `extern` wont be possible but we could have a syntax _like_ it.

**Comment by xldenis** [comment]
Closed by #141
